rust/src/delta.rs

```diff
@@ -1034,6 +1057,30 @@ impl DeltaTable {
         })
     }
 
+    pub async fn create(
+        &mut self, 
+        metadata: DeltaTableMetaData, 
+        protocol: action::Protocol
+    ) -> Result<(), DeltaTableError> {
+        // need to check if table already exists so that protocal and metaData arent overwritten on second create
+        
+        let meta = action::MetaData::try_from(metadata).unwrap();
+
+        let mut transaction = self.create_transaction(None);
+
+        //add commit info action
+        transaction.add_actions(
+            vec![
+                Action::protocol(protocol), 
+                Action::metaData(meta)
+            ]
+        );
+        transaction.commit(None).await.unwrap();
```

Comment:
it might be better to use lower level transaction API here because `commit` calls `try_commit_loop`, which is mostly designed for committing transactions into existing tables. For example, in the commit loop, it has a retry logic to keep retrying the commit with newer versions. This is not a valid action for table creation because we should only try to commit with version `0`. If we got an already exists error, we should just abort and return an error back to caller.

`try_commit_transaction` is probably what you want to use here.

Reply:
I have tried it now with the `try_commit_transaction` and it works well, however, I want to update the DeltaTable, I currently am not generating a _last_checkpoint (should that happen after a creation?) and my DeltaTable does not have a DeltaTableState, not sure how to create one from inside this function

Reply:
You don't need to worry about checkpoint here, we only create checkpoint for every 10 commits and it will be handled by https://github.com/delta-io/delta-rs/tree/main/aws/delta-checkpoint externally.

You don't need to create a new table state here since it's created in `DeltaTable::new` already. You can mutate it using `self.state` by applying the committed actions with `process_action`.

Reply:
I'm trying to update the tables state inside the `create()` but after the transaction is committed. Since I am trying to use the same actions that were just committed to use in the `process_action()`, it results in an error due to `actions: Vec<>` already being used. Should I add the `copy` or `clone` trait for the `transaction actions` or would there be a better way to handle this?

Reply:
Yes, you need to clone those actions since `add_action` takes ownership of the input.